### DISCUSSION TITLE
Preserve URL.searchParams / queryParams in redirects

### DISCUSSION THREAD

**Discussion by xriter** [withastro/roadmap#669] [Proposal]
As discussed earlier in this issue: https://github.com/withastro/astro/issues/7890 

In the current version of Redirects, the potential `URL.searchParams` don't get transferred over.

So, when you have:
```astro
// astro.config.mjs
export default defineConfig({
  redirects: {
    '/find': '/search',
  },
});
```

I would expect `/find?q=myquery` to be redirected to `/search?q=myquery`.
However, it redirects to `/search`, and the searchParams get lost.

In the issue mentioned, @ematipico and @natemoo-re made some interesting suggestions on how this desired behaviour could be implemented.

> - people might want to move a query parameter in the new URL
> ```astro
>  // astro.config.mjs
>  export default defineConfig({
>    redirects: {
>      '/find?<search-term>': '/search/<search-term>',
>    },
>  });
>```
> - people might not want to pass the query parameter in the new URL
> 
> - people might want to change the name of the new query parameter
> ```astro
>  // astro.config.mjs
>  export default defineConfig({
>    redirects: {
>      '/find': {
>  		destination: '/search',
>  		queryParams: {
>  			query: "search-term" // will change ?query=something to ?search-param=something
>  		}
>  	}	
>    },
>  });
> ```

**Comment by kamaladenalhomsi** [comment]
This should definitely be implemented in astro at some point! There are many use cases where url search params have to be preserved.
